pyerrors/input/sfcf.py

```diff
@@ -236,10 +344,84 @@ def read_sfcf(path, prefix, name, quarks='.*', corr_type='bi', noffset=0, wf=0,
             check_idl(idl[r], che[r])
         if not silent:
             print("Done")
-    result = []
-    for t in range(T):
-        result.append(Obs(deltas[t], new_names, idl=idl))
-    return result
+
+    result_dict = {}
+    if keyed_out:
+        for key in needed_keys:
+            result = []
+            for t in range(intern[name]["T"]):
+                result.append(Obs(internal_ret_dict[key][t], new_names, idl=idl))
+            result_dict[key] = result
+    else:
+        for name in name_list:
+            result_dict[name] = {}
+            for quarks in quarks_list:
+                result_dict[name][quarks] = {}
+                for off in noffset_list:
+                    result_dict[name][quarks][off] = {}
+                    for w in wf_list:
+                        result_dict[name][quarks][off][w] = {}
+                        for w2 in wf2_list:
+                            key = _specs2key(name, quarks, off, w, w2)
+                            result = []
+                            for t in range(intern[name]["T"]):
+                                result.append(Obs(internal_ret_dict[key][t], new_names, idl=idl))
+                            result_dict[name][quarks][str(off)][str(w)][str(w2)] = result
+    return result_dict
+
+
+def _lists2key(*lists):
+    sep = "/"
```

Comment:
I noticed that you define the separator char in two places.

Reply:
Oh yes, you are right... That comes from trying somewhere else, then copying...